### PR TITLE
manager: ignores references whose target column is a foreign key itself

### LINKED ISSUE #3251: Generated code "parameter isn't a function or method and can't be invoked."
**Describe the bug**

Code generation generates the following

```dart
  $$ShotTracingTableTableProcessedTableManager
      get InteractiveMapHolesTournamentId {
    final manager =
        $$ShotTracingTableTableTableManager($_db, $_db.shotTracingTable)
            .filter((f) => f.tournamentId.tournamentId($_item.tournamentId));

    final cache = $_typedResult
        .readTableOrNull(_InteractiveMapHolesTournamentIdTable($_db));
    return ProcessedTableManager(
        manager.$state.copyWith(prefetchedData: cache));
  }

```

Which results in

```dart
lib/database/database.g.dart:67382:55: Error: 'tournamentId' isn't a function or method and can't be invoked.
            .filter((f) => f.tournamentId.tournamentId($_item.tournamentId));
                                                      ^^^^^^^^^^^^
lib/database/database.g.dart:67787:38: Error: 'tournamentId' isn't a function or method and can't be invoked.
        .filter((f) => f.tournamentId($_item.tournamentId!));
                                     ^^^^^^^^^^^^
Target kernel_snapshot_program failed: Exception
```

Our table looks like 

```dart
@DataClassName('ShotTracingEntity')
class ShotTracingTable extends Table {
  @ReferenceName('InteractiveMapHolesTournamentId')
  TextColumn get tournamentId =>
      text().references(InteractiveMapHolesTable, #tournamentId)();

  IntColumn get roundNumber => integer()();

  TextColumn get playerId => text()();

  @ReferenceName('InteractiveMapHolesHoleId')
  IntColumn get holeNumber =>
      integer().references(InteractiveMapHolesTable, #holeId)();

  IntColumn get strokeNumber => integer()();

  TextColumn get clubUsed => text()();

  TextColumn get description => text().nullable()();

  TextColumn get strokeType => text()();

  RealColumn get restingPointX => real().nullable()();

  RealColumn get restingPointY => real().nullable()();

  RealColumn get restingPointZ => real().nullable()();

  RealColumn get distanceToPin => real().nullable()();

  RealColumn get distanceTravelled => real().nullable()();

  RealColumn get tempC => real().nullable()();

  RealColumn get windSpeed => real().nullable()();

  IntColumn get windDirectionDegrees => integer().nullable()();

  TextColumn get windDirection => text()();

  RealColumn get humidity => real().nullable()();

  BoolColumn get isAroundGreen =>
      boolean().withDefault(const Constant(false))();

  BoolColumn get inWaterHazard =>
      boolean().withDefault(const Constant(false))();

  BoolColumn get inTheHole => boolean().withDefault(const Constant(false))();

  @override
  Set<Column> get primaryKey => {
        tournamentId,
        roundNumber,
        playerId,
        holeNumber,
        strokeNumber,
      };
}

```

We are using latest version, and latest version of the build runner.

### PR DESCRIPTION
**Pull request by dickermoshe**
1. Closes https://github.com/simolus3/drift/issues/3251
2. Rebuild all Projects

### CODE REVIEW

**Comment by dickermoshe** [comment]
@simolus3 
It seems CI is broken...
` Invalid argument(s): Failed to load dynamic library 'libsqlite3.so': libsqlite3.so: cannot open shared object file: No such file or directory`

File: drift/test/generated/todos.dart [hunk 1]
```diff
@@ -288,6 +288,24 @@ const uuidType = DialectAwareSqlType<UuidValue>.via(
   overrides: {SqlDialect.postgres: NativeUuidType()},
 );
 
+/// Tables to test that foreign keys which target foreign keys
+/// columns are handled correctly in manager code
+class FkToPk0 extends Table {
```
Comment (simolus3) [inline drift/test/generated/todos.dart]: I prefer these as a unit test somewhere in `drift_dev` that asserts that the generated code doesn't include the relevant snippet (using `emulateDriftBuild` + `decodedMatches(isNot(contains(...)))`.

**Comment by simolus3** [comment]
The CI should work again on `develop`.

File: drift/test/generated/todos.dart (hunk 1 above)
Comment (dickermoshe) [reply drift/test/generated/todos.dart]: Done

**Review by simolus3 — APPROVED** [review]
Thanks!

**Pull request merged**